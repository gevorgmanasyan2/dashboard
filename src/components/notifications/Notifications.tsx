import { RightCircleOutlined, SendOutlined } from '@ant-design/icons'
import React, { FC, Fragment, useState } from 'react'
import CasiaDevices from '../casia-devices/casia-devices-popup/CasiaDevicesPopup'
import Popup from '../Popup/Popup'
import './notification.scss'

interface NotificationProps {
  className: string
  visible: boolean
  closeModal: () => void
}

const noteData = [
  {
    id: 1,
    icon: <SendOutlined />,
    message: 'You have a new, unconfigured Casia 360 device. Configure it now',
    status: 'unRead',
    date: 'One day ago',
  },
  {
    id: 2,
    icon: <RightCircleOutlined />,
    message: 'Your Iris Agent has left you a message: Please upload special image for',
    status: 'readed',
    date: '3 days ago',
  },
  {
    id: 3,
    icon: <SendOutlined />,
    message: 'You have a new, unconfigured Casia 360 device. Configure it now',
    status: 'readed',
    date: '3 days ago',
  },
  {
    id: 4,
    icon: <RightCircleOutlined />,
    message: 'Your Iris Agent has left you a message: Please upload special image for',
    status: 'unRead',
    date: 'One week ago',
  },
  {
    id: 5,
    icon: <SendOutlined />,
    message: 'You have a new, unconfigured Casia 360 device. Configure it now',
    status: 'readed',
    date: '3 weeks ago',
  },
]

const Notification: FC<NotificationProps> = ({ visible, closeModal }) => {
  const [isCasiaOpen, setIsCasiaOpen] = useState<boolean>(false)

  const openCasiaModal = () => {
    setIsCasiaOpen(true)
    closeModal()
  }

  const closeCasiaModal = () => setIsCasiaOpen(!isCasiaOpen)
  return (
    <div className="notification__container">
      <Popup
        className="notification__container__modal"
        visible={visible}
        closeModal={closeModal}
        footer={null}
        modalTitle="Notifications"
      >
        {noteData.map((item) => (
          <Fragment key={item.id}>
            <div
              className="notification__container__modal__content"
              onClick={() => openCasiaModal()}>
              <div className="notification__container__modal__content--icon">{item.icon}</div>
              <div className="notification__container__modal__content--message">
                <p className={item.status === 'unRead' ? 'bold' : 'default'}>{item.message}</p>
              </div>
              <div
                className={
                  item.status === 'unRead' ? 'notification__container__modal__content--status' : ''
                }
              ></div>
            </div>
            <div className="notification__container__modal__content--date">{item.date}</div>
          </Fragment>
        ))}
      </Popup>
      <CasiaDevices
        className=""
        visible={isCasiaOpen}
        closeModal={closeCasiaModal}
        okText="Cancel"
        cancelText="OK"
      />
    </div>
  )
}

export default Notification
