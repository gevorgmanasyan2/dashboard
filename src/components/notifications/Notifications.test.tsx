/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import Notifications from './Notifications'

it('should render Notifications component', () => {
  const mockItem = {
    visible: true || false,
    className: 'notification__container__modal',
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    closeModal: () => {},
  }
  expect(shallow(<Notifications {...mockItem} />)).toMatchSnapshot()
})
