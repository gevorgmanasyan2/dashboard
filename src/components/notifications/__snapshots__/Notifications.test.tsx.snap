// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Notifications component 1`] = `
<div
  className="notification__container"
>
  <Popup
    className="notification__container__modal"
    closeModal={[Function]}
    footer={null}
    modalTitle="Notifications"
    visible={true}
  >
    <div
      className="notification__container__modal__content"
      onClick={[Function]}
    >
      <div
        className="notification__container__modal__content--icon"
      >
        <ForwardRef(SendOutlined) />
      </div>
      <div
        className="notification__container__modal__content--message"
      >
        <p
          className="bold"
        >
          You have a new, unconfigured Casia 360 device. Configure it now
        </p>
      </div>
      <div
        className="notification__container__modal__content--status"
      />
    </div>
    <div
      className="notification__container__modal__content--date"
    >
      One day ago
    </div>
    <div
      className="notification__container__modal__content"
      onClick={[Function]}
    >
      <div
        className="notification__container__modal__content--icon"
      >
        <ForwardRef(RightCircleOutlined) />
      </div>
      <div
        className="notification__container__modal__content--message"
      >
        <p
          className="default"
        >
          Your Iris Agent has left you a message: Please upload special image for
        </p>
      </div>
      <div
        className=""
      />
    </div>
    <div
      className="notification__container__modal__content--date"
    >
      3 days ago
    </div>
    <div
      className="notification__container__modal__content"
      onClick={[Function]}
    >
      <div
        className="notification__container__modal__content--icon"
      >
        <ForwardRef(SendOutlined) />
      </div>
      <div
        className="notification__container__modal__content--message"
      >
        <p
          className="default"
        >
          You have a new, unconfigured Casia 360 device. Configure it now
        </p>
      </div>
      <div
        className=""
      />
    </div>
    <div
      className="notification__container__modal__content--date"
    >
      3 days ago
    </div>
    <div
      className="notification__container__modal__content"
      onClick={[Function]}
    >
      <div
        className="notification__container__modal__content--icon"
      >
        <ForwardRef(RightCircleOutlined) />
      </div>
      <div
        className="notification__container__modal__content--message"
      >
        <p
          className="bold"
        >
          Your Iris Agent has left you a message: Please upload special image for
        </p>
      </div>
      <div
        className="notification__container__modal__content--status"
      />
    </div>
    <div
      className="notification__container__modal__content--date"
    >
      One week ago
    </div>
    <div
      className="notification__container__modal__content"
      onClick={[Function]}
    >
      <div
        className="notification__container__modal__content--icon"
      >
        <ForwardRef(SendOutlined) />
      </div>
      <div
        className="notification__container__modal__content--message"
      >
        <p
          className="default"
        >
          You have a new, unconfigured Casia 360 device. Configure it now
        </p>
      </div>
      <div
        className=""
      />
    </div>
    <div
      className="notification__container__modal__content--date"
    >
      3 weeks ago
    </div>
  </Popup>
  <CasiaDevices
    cancelText="OK"
    className=""
    closeModal={[Function]}
    okText="Cancel"
    visible={false}
  />
</div>
`;
